packages/cactus-plugin-keychain-vault/src/main/typescript/plugin-keychain-vault.ts

```diff
@@ -272,14 +305,134 @@ export class PluginKeychainVault implements IPluginWebService, IPluginKeychain {
   }
 
   async set(key: string, value: string): Promise<void> {
+    if (!this.backend) {
+      throw new Error("kv vault secret engine not supported");
+    }
     const path = this.pathFor(key);
     await this.backend.write(path, { data: { value } });
     this.prometheusExporter.setTotalKeyCounter(key, "set");
   }
 
   async delete(key: string): Promise<void> {
+    if (!this.backend) {
+      throw new Error("kv vault secret engine not supported");
+    }
     const path = this.pathFor(key);
     await this.backend.delete(path);
     this.prometheusExporter.setTotalKeyCounter(key, "delete");
   }
+
+  // transit secret engine provide cryptography as service
+  // transit related method will supported if transitSecretMountPath
+  // field is provided in IPluginKeychainVaultOptions
+  // each method will start with transit
+
+  /**
+   * @description send message digest to be signed by private key stored on vault
+   * @param digest : messages digest which need to signed
+   * @param preHashed : is digest already hashed
+   * @returns asn1 encoded signature
+   */
+  async transitSign(
+    keyName: string,
+    digest: Buffer,
+    preHashed: boolean,
+  ): Promise<Buffer> {
+    if (!this.transitBackend) {
+      throw new Error("transit vault secret engine not supported");
+    }
+    const fnTag = `${this.className}#transitSign`;
+    this.log.debug(
+      `${fnTag} sign message digest with size = ${digest.length} , preHashed = ${preHashed} with key = ${keyName}`,
+    );
+    try {
+      const resp = await this.transitBackend.write("sign/" + keyName, {
+        input: digest.toString("base64"),
+        prehashed: preHashed,
+        marshaling_algorithm: "asn1",
+      });
+      this.log.debug(`Response from Vault: %o`, JSON.stringify(resp));
+      if (resp?.data?.signature) {
+        const base64Sig = (resp.data.signature as string).split(":")[2];
+        return Buffer.from(base64Sig, "base64");
+      } else {
+        throw new Error(
+          `${fnTag}: Invalid response received from Vault. Expected "resp.data.signature" property chain to be truthy`,
+        );
+      }
+    } catch (error) {
+      if ((error as any)?.response?.statusCode === HttpStatus.NOT_FOUND) {
+        throw new Error("signing key not found");
```

Comment:
@Zzocker Include `keyName` and `fnTag` in the error message to provide more context. Further details provided below in my other comment.